string parameter
string[] sub_param
string value